Include/internal/pycore_long.h

```diff
@@ -296,7 +299,7 @@ _PyLong_FlipSign(PyLongObject *op) {
         .long_value  = { \
             .lv_tag = TAG_FROM_SIGN_AND_SIZE( \
                 (val) == 0 ? 0 : ((val) < 0 ? -1 : 1), \
-                (val) == 0 ? 0 : 1), \
+                (val) == 0 ? 0 : 1) | IMMORTAL_BIT(val), \
```

Comment:
`_PyLong_DIGIT_INIT` is only used to initialize statically allocated ints, so you can replace `IMMORTAL_BIT(val)` with IMMORTALITY_BIT_MASK.